src/irmin-pack/pack_store_intf.ml

```diff
@@ -48,11 +52,34 @@ module type Maker = sig
        and type hash = hash
        and type value = V.t
        and type index := index
+       and type indexing_strategy := indexing_strategy
 end
 
 module type Sigs = sig
-  module type S = S
-  module type Maker = Maker
+  module Indexing_strategy : sig
+    type t = value_length:int -> Pack_value.Kind.t -> bool
+    (** The type of configurations for [irmin-pack]'s indexing strategy, which
+        dictates whether or not newly-appended pack entries should also be added
+        to the index. Strategies are parameterised over:
+
+        - the length of the binary encoding of the {i object} inside the pack
+          entry (i.e. not accounting for the encoded hash and kind character);
+        - the kind of the pack object having been added.
+
+        Indexing more than the {!minimal} strategy only impacts performance and
+        not correctness: more indexing results in a larger index and a smaller
+        pack file. *)
+
+    val always : t
```

Comment:
Perhaps as well as always and minimal, have a "default" (which could be minimal, or always, or something else later) which is used when optional indexing_strategy arguments to functions are not supplied.

Reply:
Good idea; done :+1: